Assets/Scripts/Views/AssetViewer.cs

```diff
@@ -100,4 +118,55 @@ public void Lighten(GameObject prefab)
     {
         prefab.GetComponentInChildren<Image>().color = Color.white;
     }
+
+    public void MoveTo(GameObject prefab, Vector2 target, float speed, MovementTypes style)
+    {
+        moveCoroutine = StartCoroutine(MoveAsset(prefab, target, speed, style));
+    }
+    
+    //For SmoothDamp only; higher speed = slower movement
+    private IEnumerator MoveAsset(GameObject prefab, Vector2 target, float speed, MovementTypes style)
+    {
+        isMoving = true;
+        movingPrefab = prefab;
+        targetPosition = target;
+        
+        Vector2 currentPosition = prefab.transform.position;
+        Vector2 velocity = Vector2.zero;
+
+        speed *= Time.deltaTime;
+
+        while (Vector2.Distance(currentPosition, target) > .1f)
+        {
+            switch (style)
```

Comment:
Rider recommended turning this switch into this:
```csharp
currentPosition = style switch
{
    MovementTypes.Smooth => Vector2.MoveTowards(currentPosition, target, speed * 50),
    MovementTypes.FastStart => Vector2.Lerp(currentPosition, target, speed),
    MovementTypes.FastMiddle => Vector2.SmoothDamp(currentPosition, target, ref velocity, speed * 10),
     _ => currentPosition
};
```

Thoughts?

Reply:
I've never seen such a clean switch statement. I think I'm in love.